### ISSUE TITLE
I just want my Element that can only be moved to viewport，because my element is position of fixed。but I set bounds="body" and cannot work,is still can move  in the direction of down while out of viewpoint 

### ISSUE THREAD

**Issue by hututuhu** [STRML/react-draggable#714]
```
<Draggable
    disabled={!isDraggable}
    handle=".br-modal__header"
    defaultClassNameDragged="br-modal__default-drag"
    positionOffset={unblock ? { x: '-50%', y: '-50%' } : undefined}
    cancel={resizable ? '.react-resizable-handle' : undefined}
    bounds="body"
  >
    {resizeElement}
</Draggable>
```
![image](https://user-images.githubusercontent.com/37233828/226308835-836bf027-0414-44d0-beab-0c3044fe86b1.png)

**Comment by tinywaves** [comment]
Yes, I also encountered this problem. Do you have the solution now? Thank you. @hututuhu

**Comment by hardik79** [comment]
Hello @tinyRipple and @hututuhu ,
I have a solution to your problem. Please add **bounds={bounds}** into the Draggable and **ref={dragRef}** into the main draggable div like
![Screenshot_1](https://github.com/react-grid-layout/react-draggable/assets/72906505/f2248086-7334-49aa-ab7f-5ce9193b9b81)

**Comment by hardik79** [comment]
Check https://github.com/hardik79/react-draggable/blob/main/src/App.js solution example for your problem.

**Comment by hututuhu** [comment]
> Yes, I also encountered this problem. Do you have the solution now? Thank you. @hututuhu

I solve this problem by using onStop and position.
![image](https://github.com/react-grid-layout/react-draggable/assets/37233828/24f5a153-6cd9-42a7-9deb-32d86725fa33)

![image](https://github.com/react-grid-layout/react-draggable/assets/37233828/45264f42-e8cf-4106-b1c0-b3183589ce7d)

**Comment by YuLingCheng** [comment]
> Check https://github.com/hardik79/react-draggable/blob/main/src/App.js solution example for your problem.

Setting the computed bounds works well, thanks for the tip 👍 
I think we can manage without a `useEffect` and rely on the `onStart` prop to recompute the bounds whenever the users want to start dragging. I just used this strategy to solve similar issue to limit the bounds of a draggable Mui modal.

```tsx
function App() {
  const [bounds, setBounds] = useState({});
  const contentRef = useRef(null);

  function updateBounds() {
    if (contentRef.current !== null) {
      setBounds({
        left: ..., 
        top: ...,
        right: ...,
        bottom: ..., // Whatever computation works in your case based on contentRef.current
      });
    }

  return (
    <Draggable bounds={bounds} onStart={updateBounds}>
      <YourComponent ref={contentRef} />
    </Draggable>
  );
}
```
